app/backend/approaches/retrievethenread.py

```diff
@@ -122,18 +126,53 @@ async def run(
             new_user_content=user_content,
             max_tokens=self.chatgpt_token_limit - response_token_limit,
         )
-
-        chat_completion = (
-            await self.openai_client.chat.completions.create(
-                # Azure OpenAI takes the deployment name as the model name
-                model=self.chatgpt_deployment if self.chatgpt_deployment else self.chatgpt_model,
-                messages=updated_messages,
-                temperature=overrides.get("temperature", 0.3),
-                max_tokens=response_token_limit,
-                n=1,
-            )
-        ).model_dump()
-
+        
+        if(self.use_hugging_face):
+            API_URL = os.getenv("HUGGINGFACE_API_URL")
+            API_KEY = os.getenv("HUGGINGFACE_API_KEY")
+
+            headers = {"Authorization": "Bearer " + API_KEY}
+            
+            huggingf_question_prompt = ''
+            for message in updated_messages:
+                if message["content"]:
+                    huggingf_question_prompt += message["content"] + '\n'
+
+                                        
+            def ask_huggingface(payload):
+                response = requests.post(API_URL, headers=headers, json=payload)
+                return response.json()
+            
+            chat_completion = ask_huggingface({
+            "inputs": huggingf_question_prompt,
+            "parameters": {
+                "temperature" : 1.6,
+                "max_length": response_token_limit,
+                "return_full_text": False,
+                "top_k": 1,
+                "top_p": 0.8,
+                "repetition_penalty": 0.1,
+                "max_new_tokens": 250,
+            },
+            "options": {
+                "use_cache": False
+            }
+        })
+        else:     
+            chat_completion = (
+                await self.openai_client.chat.completions.create(
+                    # Azure OpenAI takes the deployment name as the model name
+                    model=self.chatgpt_deployment if self.chatgpt_deployment else self.chatgpt_model,
+                    messages=updated_messages,
+                    temperature=overrides.get("temperature", 0.3),
+                    max_tokens=response_token_limit,
+                    n=1,
+                )
+            ).model_dump()
+        
+            
+        
+        
```

Comment:
One empty line should do (again, note comment wrt running the linters/formaters)!